### ISSUE TITLE
Metadata cannot round-trip

### ISSUE THREAD

**Issue by joshwatson** [Vector35/binaryninja-api#1155]
```
>>> bv.store_metadata('buggy', {0: 12})
>>> bv.query_metadata('buggy')
{'0': 12L}
>>> bv.store_metadata('buggy2', _)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Applications/Binary Ninja.app/Contents/MacOS/plugins/../../Resources/python/binaryninja/binaryview.py", line 3574, in store_metadata
    core.BNBinaryViewStoreMetadata(self.handle, key, metadata.Metadata(md).handle)
  File "/Applications/Binary Ninja.app/Contents/MacOS/plugins/../../Resources/python/binaryninja/metadata.py", line 61, in __init__
    md = Metadata(value[elm], signed, raw)
  File "/Applications/Binary Ninja.app/Contents/MacOS/plugins/../../Resources/python/binaryninja/metadata.py", line 64, in __init__
    raise ValueError("List doesn't not contain type of: int, bool, str, float, list, dict")
ValueError: List doesn't not contain type of: int, bool, str, float, list, dict
```

**Comment by plafosse** [comment]
Because I just talked about this in chat... This maps to a native API, and thus any language which has bindings to the native API.  e have to coerce python int to C++ uint64 then to python ... well we chose to do long because its the best option, if we changed it to int it would be broken in a different way. The only way to fix it would to be to include some other bit of metadata that says what each of the values should be coerced back to for python 2.7.

**Comment by plafosse** [comment]
Because I just discussed this in chat... This is specifically a python2.7 issue that I don't think we will fix. We have to coerce python `int` to C++ `uint64` in the native API. To read it back out we have to decide how what python2 type to convert the native `uint64` into. We chose a python `long` because it can represent the full with `uint64 value.  The only way to fix it would to be to include some other bit of metadata that says what each of the values should be coerced back to for python 2.7. If anyone has a suggestion on a better way to do this I'm happy to entertain suggestions :)
